packages/sdk-react/src/core/theme/types.ts

```diff
@@ -0,0 +1,219 @@
+
+export type ThemeConfig = {
+  borderRadius?: number;
+
+  colors?: {
+    primary?: string;
+    secondary?: string;
+    neutral?: string;
+
+    background?: string;
+
+    text?: string;
+  };
+
+  typography?: {
+    fontFamily?: string;
+    fontSize?: number;
+
+    h1?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    h2?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    h3?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    subtitle1?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    subtitle2?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    body1?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+    body2?: {
+      fontSize?: string | number;
+      fontWeight?: string | number;
+      lineHeight?: string | number;
+    };
+  };
+};
+
+interface MoniteTablePaginationRootSlotProps {
+  pageSizeOptions?: number[];
+}
+
+interface MoniteTablePaginationSlotProps {
+  slotProps?: {
+    pageSizeSelect?: Omit<
+      SelectProps,
+      'value' | 'defaultValue' | 'aria-label' | 'ref' | 'components'
+    >;
+  };
+}
+
+export interface MoniteTablePaginationProps
+  extends MoniteTablePaginationSlotProps,
+    MoniteTablePaginationRootSlotProps {}
+
+export interface MoniteInvoiceStatusChipProps {
+  icon?: boolean;
+  /** The variant of the Chip. */
+  variant?: ChipProps['variant'];
+  /** The size of the Chip. */
+  size?: ChipProps['size'];
+  /** Display status icon? */
+  /** The status of the invoice. */
+  status: components['schemas']['ReceivablesStatusEnum'];
+  /** The variant of the Chip. */
+}
+
+export interface MoniteInvoiceRecurrenceStatusChipProps {
+  /** The status of the recurrence. */
+  status: components['schemas']['RecurrenceStatus'];
+  /** The variant of the Chip. */
+  variant?: ChipProps['variant'];
+  /** The size of the Chip. */
+  size?: ChipProps['size'];
+  /** Display status icon? */
+  icon?: boolean;
+}
+
+export interface MoniteInvoiceRecurrenceIterationStatusChipProps {
+  /** The status of the iteration. */
+  status: components['schemas']['IterationStatus'];
+  /** The variant of the Chip. */
+  variant?: ChipProps['variant'];
+  /** The size of the Chip. */
+  size?: ChipProps['size'];
+  /** Display status icon? */
+  icon?: boolean;
+}
+
+export interface MonitePayableStatusChipProps {
+  /** The status of the payable. */
+  status: components['schemas']['PayableStateEnum'];
+  /** Display status icon? */
+  icon?: boolean;
+  /** The variant of the Chip. */
+  variant?: ChipProps['variant'];
+  /** The size of the Chip. */
+  size?: ChipProps['size'];
+}
```

Comment:
Can we explore creating a base type and extending each unique type to avoid repetition for the ChipProps?

Reply:
updated, check pls